Fix  project('shared') javadoc errors or warnings

**Problem description**
Error Logs in shared javadocs 
```
Successfully started process 'command '/usr/lib64/jvm/java-1.8.0-oracle-1.8.0.111/bin/javadoc''
/var/jenkins_home/workspace/pravega/shared/src/main/java/com/emc/pravega/shared/metrics/Counter.java:39: warning: no @return
    long get();
         ^
/var/jenkins_home/workspace/pravega/shared/src/main/java/com/emc/pravega/shared/metrics/Counter.java:44: warning: no @return
    String getName();
           ^
/var/jenkins_home/workspace/pravega/shared/src/main/java/com/emc/pravega/shared/metrics/StatsLogger.java:47: warning: no @return
    <T extends Number> Gauge registerGauge(String name, Supplier<T> value);
                             ^
/var/jenkins_home/workspace/pravega/shared/src/main/java/com/emc/pravega/shared/metrics/StatsProvider.java:35: warning: no @return
    DynamicLogger createDynamicLogger();
```

**Problem location**
build.gradle -> javadoc 

**Suggestions for an improvement**
suppress `shared/` when running javadoc

--- a/build.gradle
+++ b/build.gradle
@@ -181,6 +181,9 @@ project ('shared') {
             }
         }
     }
+    javadoc {
+        exclude 'com/emc/pravega/shared/*'
+    }
 }
 
 project('clients:streaming') {

